From 3f94a979eebd1c9496b49b4e07b7823550dec97e Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Wed, 23 Aug 2017 17:06:20 +0200
Subject: [PATCH 74/93] localauth plugin: change return code of sss_an2ln
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

It is expected that the an2ln plugin function returns KRB5_LNAME_NOTRANS
to indicate that no mapping can be determined and other an2ln methods
can be tried. Currently SSSD's localauth plugin returns
KRB5_PLUGIN_NO_HANDLE which sould only be used for the userok plugin
function.

Resolves https://pagure.io/SSSD/sssd/issue/3459

Reviewed-by: Jakub Hrozek <jhrozek@redhat.com>
Reviewed-by: Fabiano Fidêncio <fidencio@redhat.com>
---
 src/krb5_plugin/sssd_krb5_localauth_plugin.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/krb5_plugin/sssd_krb5_localauth_plugin.c b/src/krb5_plugin/sssd_krb5_localauth_plugin.c
index 13ab07d1315430f11f7bbc916f25d2c837bec78c..60567d783949b0e8b5537f6d662ff392dc848b32 100644
--- a/src/krb5_plugin/sssd_krb5_localauth_plugin.c
+++ b/src/krb5_plugin/sssd_krb5_localauth_plugin.c
@@ -137,7 +137,7 @@ static krb5_error_code sss_an2ln(krb5_context context,
                                      &nss_errno);
     if (nss_status != NSS_STATUS_SUCCESS) {
         if (nss_status == NSS_STATUS_NOTFOUND) {
-            ret = KRB5_PLUGIN_NO_HANDLE;
+            ret = KRB5_LNAME_NOTRANS;
         } else {
             ret = EIO;
         }
-- 
2.14.1

